From 1f699430806ce304a048738c353b1cff15f238bb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 10 Apr 2017 00:33:08 +0100
Subject: [PATCH] gdm: Allow empty password text

While this is a nice UI feature to have, it isn't actually all that useful.
Furthermore, livecds are designed to be passwordless, so not allowing an
empty password for the login is unhelpful.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 js/gdm/authPrompt.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/js/gdm/authPrompt.js b/js/gdm/authPrompt.js
index 994c080..2e59ee2 100644
--- a/js/gdm/authPrompt.js
+++ b/js/gdm/authPrompt.js
@@ -181,14 +181,14 @@ const AuthPrompt = new Lang.Class({
                               x_align: St.Align.END,
                               y_align: St.Align.END });
 
-        this._updateNextButtonSensitivity(this._entry.text.length > 0);
+        this._updateNextButtonSensitivity(true);
 
         this._entry.clutter_text.connect('text-changed',
                                          Lang.bind(this, function() {
                                              if (!this._userVerifier.hasPendingMessages)
                                                  this._fadeOutMessage();
 
-                                             this._updateNextButtonSensitivity(this._entry.text.length > 0);
+                                             this._updateNextButtonSensitivity(true);
                                          }));
         this._entry.clutter_text.connect('activate', Lang.bind(this, function() {
             if (this.nextButton.reactive)
@@ -416,7 +416,7 @@ const AuthPrompt = new Lang.Class({
     },
 
     updateSensitivity: function(sensitive) {
-        this._updateNextButtonSensitivity(sensitive && this._entry.text.length > 0);
+        this._updateNextButtonSensitivity(sensitive);
         this._entry.reactive = sensitive;
         this._entry.clutter_text.editable = sensitive;
     },
-- 
2.12.2

